File: Review-04-Python-Unit-Testing/Example-1/tests/examples/test_board.py
import copy

import pytest
from hamcrest import *

from examples.board import Board

EXPECTED_EMPTY_STR = "1|2|3\n4|5|6\n7|8|9"


@pytest.fixture
def a_board():
    yield Board()


def test_default_constructor(a_board):
    for expected_char in range(1, 10):
        assert_that(a_board.get_cell(expected_char), is_(str(expected_char)))

    assert_that(str(a_board), equal_to(EXPECTED_EMPTY_STR))
    assert_that(not a_board.is_full())

    retrieved = a_board.get_3_cells(1, 2, 3)
    expected = ((0, "1"), (1, "2"), (2, "3"))

    assert_that(retrieved[0], equal_to(expected[0]))
    assert_that(retrieved[1], equal_to(expected[1]))
    assert_that(retrieved[2], equal_to(expected[2]))


def test_set_cell(a_board):
    a_board.set_cell(1, "X")
    a_board.set_cell(9, "O")

    assert_that(a_board.get_cell(1), is_("X"))
    assert_that(a_board.get_cell(9), is_("O"))

    retrieved = a_board.get_3_cells(1, 5, 9)
    expected = ((0, "X"), (4, "5"), (8, "O"))

    assert_that(retrieved[0], equal_to(expected[0]))
    assert_that(retrieved[1], equal_to(expected[1]))
    assert_that(retrieved[2], equal_to(expected[2]))

    assert_that(str(a_board), is_not(equal_to(EXPECTED_EMPTY_STR)))
    assert_that(str(a_board), equal_to("X|2|3\n4|5|6\n7|8|O"))

    assert_that(not a_board.is_full())

File: Review-15-Java-Strategy-Pattern/Example-4/src/test/java/edu/odu/cs/cs330/examples/tictactoe/TestBoard.java
package edu.odu.cs.cs330.examples.tictactoe;

import org.junit.jupiter.api.TestMethodOrder;
import org.junit.jupiter.api.MethodOrderer;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.BeforeEach;

import static org.junit.jupiter.api.Assertions.*;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.*;
// import org.hamcrest.core.IsNull;

/**
 * 1 - Does this piece of code perform the operations
 *     it was designed to perform?
 *
 * 2 - Does this piece of code do something it was not
 *     designed to perform?
 *
 * 1 Test per mutator
 */
@TestMethodOrder(MethodOrderer.MethodName.class)
public class TestBoard
{
    private static final String EXPECTED_EMPTY_STRING = "1|2|3\n4|5|6\n7|8|9\n";

    Board aBoard;

    @BeforeEach
    public void setUp()
    {
        aBoard = new Board();
    }

    @Test
    public void testDefaultConstructor()
    {
        for (int i = 1; i <= 9; i++) {
            char expectedChar = Character.forDigit(i, 10);
            assertThat(aBoard.getCell(i), is(expectedChar));
        }

        assertThat(aBoard.toString(), equalTo(EXPECTED_EMPTY_STRING));
        assertFalse(aBoard.isFull());

        char[] retrieved = aBoard.get3Cells(1, 2, 3);

        assertThat(retrieved[0], is(equalTo('1')));
        assertThat(retrieved[1], is(equalTo('2')));
        assertThat(retrieved[2], is(equalTo('3')));
    }

    @Test
    public void testSetCell()
    {
        aBoard.setCell(1, 'X');
        aBoard.setCell(9, 'O');

        assertThat(aBoard.getCell(1), is('X'));
        assertThat(aBoard.getCell(9), is('O'));

        char[] retrieved = aBoard.get3Cells(1, 5, 9);

        assertThat(retrieved[0], is(equalTo('X')));
        assertThat(retrieved[1], is(equalTo('5')));
        assertThat(retrieved[2], is(equalTo('O')));

        assertThat(aBoard.toString(), not(equalTo(EXPECTED_EMPTY_STRING)));
        assertThat(aBoard.toString(), equalTo("X|2|3\n4|5|6\n7|8|O\n"));

        assertFalse(aBoard.isFull());
    }

}
